segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageTests.java

```diff
@@ -2561,6 +2561,53 @@ private void testParallelReadRequestsOnSingleSegmentWithReentry(int numberOfRequ
         CompletableFuture.allOf(futures).join();
     }
 
+    /**
+     * Test concurrent writes to storage system segments by simulating concurrent writes.
+     *
+     * @throws Exception Throws exception in case of any error.
+     */
+    @Test
+    public void testSystemSegmentConcurrency() throws Exception {
+
+        SegmentRollingPolicy policy = new SegmentRollingPolicy(2); // Force rollover after every 2 byte.
+        @Cleanup
+        TestContext testContext = getTestContext(ChunkedSegmentStorageConfig.DEFAULT_CONFIG.toBuilder().indexBlockSize(3).build());
+        if (!(testContext.metadataStore instanceof InMemoryMetadataStore)) {
```

Comment:
Why are we doing this? This will silently not run the test at all for pretty much non-InMemoryMetadataStore.

Is there a specific unit test class/suite that uses the InMemory one? If so, let's add this test there.

Reply:
removed this check. This test does not have any special test hooks that would fail in non-InMemoryMetadataStore-.